src/pke/lib/scheme/bfvrns/bfvrns-leveledshe.cpp

```diff
@@ -77,8 +89,20 @@ void LeveledSHEBFVRNS::EvalSubInPlace(
   std::vector<DCRTPoly> &cv = ciphertext->GetElements();
 
   DCRTPoly pt = plaintext->GetElement<DCRTPoly>();
-  pt.SetFormat(EVALUATION);
-  cv[0] -= pt.Times(cryptoParams->GetQDivtModq());
+  if (cryptoParams->GetEncryptionTechnique() == POVERQ) {
+    pt.SetFormat(COEFFICIENT);
+    const NativeInteger &NegQModt = cryptoParams->GetNegQModt();
+    const NativeInteger &NegQModtPrecon = cryptoParams->GetNegQModtPrecon();
+    const std::vector<NativeInteger> &tInvModq = cryptoParams->GettInvModq();
+    const std::vector<NativeInteger> &tInvModqPrecon = cryptoParams->GettInvModqPrecon();
+    const NativeInteger t = cryptoParams->GetPlaintextModulus();
+    pt.TimesQovert(cryptoParams->GetElementParams(), tInvModq, tInvModqPrecon, t, NegQModt, NegQModtPrecon);
+    pt.SetFormat(EVALUATION);
```

Comment:
can you pull "pt.SetFormat(EVALUATION);" out of this if statement and add it after?